src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumStorageController.java

```diff
@@ -0,0 +1,512 @@
+            for (EnumFacing facing : EnumFacing.VALUES) {
+                BlockPos offsetPos = pos.offset(facing);
+                if (checked.contains(offsetPos) || getPos().equals(offsetPos)) continue;
+                if (getWorld().getBlockState(pos).getBlock() == Blocks.AIR) continue;
+
+                // add a new pos to search
+                if (checkStorageNeighbor(mte, facing))
+                    searchQueue.add(offsetPos);
+            }
+        }
+
+        // check old posses to disconnect the storages
+        for (BlockPos pos : oldPositions) {
+
+            // if we already checked this pos before, don't check it again
+            if (checked.contains(pos)) continue;
+
+            // if the pos is air, there's nothing to check
+            if (getWorld().getBlockState(pos).getBlock() == Blocks.AIR) continue;
+
+            IQuantumStorage<?> storage = oldInstances.get(pos).get();
+            if (storage == null) {
+                MetaTileEntity mte = GTUtility.getMetaTileEntity(getWorld(), pos);
+                if (!(mte instanceof IQuantumStorage<?>quantumStorage)) {
+                    continue;
+                }
+                storage = quantumStorage;
+            }
+            storage.setDisconnected();
+        }
+        handler.rebuildCache();
+        calculateEnergyUsage();
+        markDirty();
+    }
+
+    private static boolean checkStorageNeighbor(MetaTileEntity mte, EnumFacing facing) {
+        if (mte.getNeighbor(facing) instanceof IGregTechTileEntity gtte) {
+            return gtte.getMetaTileEntity() instanceof IQuantumStorage<?>;
+        }
+        return false;
+    }
+
+    private void calculateEnergyUsage() {
+        energyContainers.clear();
+        energyConsumption = 0;
+        for (var pos : storagePositions) {
+            var storage = getStorage(pos);
+            if (storage != null) {
+                energyConsumption += getTypeEnergy(storage);
+                if (storage.getType() == IQuantumStorage.Type.ENERGY) {
+                    energyContainers.add((IEnergyContainer) storage.getTypeValue());
+                }
+            }
+        }
+        energyHandler = new EnergyContainerList(energyContainers);
+        writeCustomData(GregtechDataCodes.UPDATE_ENERGY_PER, buf -> buf.writeLong(energyConsumption));
+    }
+
+    private long getTypeEnergy(IQuantumStorage<?> storage) {
+        return switch (storage.getType()) {
+            case ITEM, FLUID -> {
+                int tier = storage instanceof ITieredMetaTileEntity tieredMTE ? tieredMTE.getTier() : 1;
+                yield tier > 5 ?
+                        GTValues.V[GTValues.HV] / 2 :
+                        GTValues.V[GTValues.LV] / 2;
+            }
+            case PROXY -> 8L;
+            case EXTENDER -> 2L;
+            case ENERGY -> 1L;
+        };
+    }
+
+    private void addEnergy(IQuantumStorage<?> storage) {
```

Comment:
This and the following methods are not used at all. Were they being used at one point?

Reply:
there were being used to add stuff to the controller like a list, but i forgot to remove them